doc/shared/sunadaptcontroller/SUNAdaptController_Description.rst

```diff
@@ -0,0 +1,502 @@
+
+The base ``SUNAdaptController`` class is modeled after SUNDIALS' other object-oriented
+classes, in that this class contains a pointer to an implementation-specific
+*content*, an *ops* structure with generic controller operations, and a
+:c:type:`SUNContext` object. Specifically, the type ``SUNAdaptController`` is defined
+as:
+
+.. c:type:: struct _generic_SUNAdaptController *SUNAdaptController
+
+and the base class structure is defined as
+
+.. code-block:: C
+
+   struct _generic_SUNAdaptController {
+        void* content;
+        generic_SUNAdaptController_Ops_* ops;
+        SUNContext sunctx;
+    };
+
+Here, ``_generic_SUNAdaptController_Ops`` is the pointer to a structure containing
+function pointers to the various controller operations, and is defined as
+
+.. code-block:: c
+
+    struct _generic_SUNAdaptController_Ops {
+        SUNAdaptController_Type (*getid)(SUNAdaptController C);
+        int (*destroy)(SUNAdaptController C);
+        int (*estimatestep)(SUNAdaptController C, sunrealtype h, sunrealtype dsm, sunrealtype* hnew);
+        int (*estimatestepandorder)(SUNAdaptController C, sunrealtype h, int q, sunrealtype dsm, sunrealtype* hnew, int *qnew);
+        int (*estimatemristeps)(SUNAdaptController C, sunrealtype H, sunrealtype h, sunrealtype DSM, sunrealtype dsm, sunrealtype* Hnew, sunrealtype *hnew);
+        int (*estimatesteptol)(SUNAdaptController C, sunrealtype H, sunrealtype tolfac, sunrealtype DSM, sunrealtype dsm, sunrealtype *Hnew, sunrealtype* tolfacnew);
+        int (*reset)(SUNAdaptController C);
+        int (*setdefaults)(SUNAdaptController C);
+        int (*write)(SUNAdaptController C, FILE* fptr);
+        int (*setmethodorder)(SUNAdaptController C, int p, int q);
+        int (*adjustcontrollerorder)(SUNAdaptController C, int adj);
+        int (*seterrorbias)(SUNAdaptController C, sunrealtype bias);
+        int (*update)(SUNAdaptController C, sunrealtype h, sunrealtype dsm);
+        int (*updatemrih)(SUNAdaptController C, sunrealtype H, sunrealtype h, sunrealtype DSM, sunrealtype dsm);
+        int (*updatemritol)(SUNAdaptController C, sunrealtype H, sunrealtype tolfac, sunrealtype DSM, sunrealtype dsm);
+        int (*space)(SUNAdaptController C, long int *lenrw, long int *leniw);
+    };
+
+
+.. _SUNAdaptController.Description.controllerTypes:
+
+SUNAdaptController Types
+------------------------
+
+The time integrators in SUNDIALS adapt a variety of parameters to achieve
+accurate and efficient computations. To this end, each SUNAdaptController implementation
+should note its type, so that integrators will understand the types of
+adaptivity that the controller is designed to perform. These are encoded in the
+following set of SUNAdaptController types:
+
+.. c:enum:: SUNAdaptController_Type
+
+   The enumerated type :c:type:`SUNAdaptController_Type` defines the enumeration
+   constants for SUNDIALS error controller types
+
+.. c:enumerator:: SUN_ADAPTCONTROLLER_NONE
```

Comment:
Is this type necessary? Can't we implement a `SUN_ADAPTCONTROLLER_H` that does not change the value of `h`, i.e., an identity controller.